src/bufr/BufrParser/BufrParser.cpp

```diff
@@ -139,18 +143,37 @@ namespace Ingester
         return splitDataMap;
     }
 
-    void BufrParser::openBufrFile(const std::string &filepath)
+    void BufrParser::openBufrFile(const std::string& filepath,
+                                  bool isStdFormat,
+                                  const std::string& tablepath)
     {
         fortranFileId_ = 11;  // Fortran file id must be a integer > 10
         open_f(fortranFileId_, filepath.c_str());
-        openbf_f(fortranFileId_, "IN", fortranFileId_);
+
+        if (!isStdFormat)
+        {
+            openbf_f(fortranFileId_, "IN", fortranFileId_);
+        }
+        else
+        {
+            openbf_f(fortranFileId_, "SEC3", fortranFileId_);
+
+            if (!tablepath.empty())  // else use the default tables
+            {
+                table1FileId_ = fortranFileId_ + 1;
+                table2FileId_ = fortranFileId_ + 2;
+                mtinfo_f(tablepath.c_str(), table1FileId_, table2FileId_);
+            }
+        }
```

Comment:
I am not loving this `if-else` business all over the place.

Reply:
@aerorahul This is quiet literally the only place in the program where we check isStdFormat and do anything with that information (I don't know what you mean by "all over the place"). WMO BUFR have a slightly different procedure to open the files which we handle succinctly in this one if statement which gets called exactly once, otherwise there is no difference at all with how you deal with these files. The altrernative would be orders of magnitude more complex. It would mean to create a BufrParser, NcepBufrParser and WMOBufrParser classes (the two new classes would do nothing other than open the file in a slightly different way), but then you'd still need an if statement in order to instantiate the correct flavor based on the contents of the YAML file. So how do you work around that? Well we add a factory to construct the correct version of the BufrParser based on the information we get from the YAML file. So now you've created at least 2 classes and a full fledged factory just to avoid 1 if statement.

It would make sense to have different BufrParser subclasses if these files needed to be treated differently and the code if (isStdFormat) appeared in multiple places, but we just don't.